### DISCUSSION TITLE
PolarFire SoC Icicle Kit MTD Peripheral Test - POLARFIRE-SOC

### DISCUSSION THREAD

**Discussion by gsriker77** [polarfire-soc/polarfire-soc-documentation#219] [Reference designs, FPGA and development kits] [answered]
We have 2x PolarFire SoC Icicle Kits for evaluation of the product. When connecting the UART connections and powering up the board we get the PolarFire SoC Icicle Kit MTD Peripheral Test v2022.04-RC2 running automatically.
It might help us figure out the 'non-debug' startup location on these boards just for learning purposes.

I don't get the boot log on COM0 because the MTD is STILL LOADED ON THE BOARD
Where is this test programmed into on the board/SoC...i.e.: where is it running from? eMMC, eNVM????

Thanks

**Comment by hughbreslin** [comment]
Can you connect the debugger in the "attach" configuration for any bare metal example and view the program counter address to determine the memory location being used? I am not sure of the answer but this would be how I check.

**Comment by gsriker77** [comment]
These are new boards received early summer this year.

Regarding the other question: 'non-debug' meaning I'm not using the On-chip debugger to load/run a program from LIM.  

So when I power up the Icicle Kit board, what is the 'Reset Vector' Address for the MSS.  Every processor in the world has a starting address when they get released from reset.  I've still not found the Microsemi/Microchip document that describes what that address is.  For example x86 processor reset vector is 0xFFFF0, ARM processor reset is 0x0 or 0xFFFF0000, RISC-V E51/U54 core in the SoC MSS reset vector is ?????

If you find that information (expecting you to), which document/link/reference was this information obtained from,

**Comment by gsriker77** [comment, marked as answer]
https://github.com/polarfire-soc/polarfire-soc-documentation/blob/master/fundamentals/boot-modes/boot-mode-0/boot-mode-0-fundamentals.md
This has information on Reset vector. Please read the System registers information along with this doc

**Comment by gsriker77** [comment]
The above link helps. Better if it include as part of TRM

Thanks

**Comment by hughbreslin** [comment]
Hi @gsriker77 I'm just going to respond to the messages above all in one thread:

> Regarding the other question: 'non-debug' meaning I'm not using the On-chip debugger to load/run a program from LIM.

I recommended an "attach" configuration as this will allow you to see the current execution address of the harts without resetting them or downloading an additional application. 

> So when I power up the Icicle Kit board, what is the 'Reset Vector' Address for the MSS. Every processor in the world has a starting address when they get released from reset. I've still not found the Microsemi/Microchip document that describes what that address is. For example x86 processor reset vector is 0xFFFF0, ARM processor reset is 0x0 or 0xFFFF0000, RISC-V E51/U54 core in the SoC MSS reset vector is ?????

The reset vector isn't fixed for MPFS (as we have a different reset vector for different boot modes as you saw in the documentation). Depending on where the application is running from you should see a different execution address in the program counter when you connect. You could also do an initial reset without the "continue" option selected and see where the start address is too. 

> The above link helps. Better if it include as part of TRM

Have you opened a ticket to include this information in the TRM?

**Comment by gsriker77** [comment]
Yes, FD-217 
i.e. to add relevant info in TRM.

Thanks
